### ISSUE TITLE
expected arguments: *CudaTensor~2D* [CudaTensor~2D] [float] CudaTensor~2D CudaTensor~2D

### ISSUE THREAD

**Issue by sahasuman** [cmusatyalab/openface#203]
The following is my network model:

local function create_model(opt) 
  local name = 'frcnnSS_alexnet'
  backend = backend or cudnn

  -- SHARED PART
  local shared   = nn.Sequential()
  local conv1 =backend.SpatialConvolution(3,96,11,11,4,4,5,5,1)
  conv1.name = 'conv1'
  -- Freeze conv1
  conv1.accGradParameters = function() end
  shared:add(conv1)
  shared:add(backend.ReLU(true))
  shared:add(backend.SpatialMaxPooling(3,3,2,2,1,1):ceil())
  shared:add(inn.SpatialCrossResponseNormalization(5,0.0001,0.75,1))

  local conv2 = backend.SpatialConvolution(96,256,5,5,1,1,2,2,2)
  conv2.name = 'conv2'
  shared:add(conv2)
  shared:add(backend.ReLU(true))
  shared:add(backend.SpatialMaxPooling(3,3,2,2,1,1):ceil())
  shared:add(inn.SpatialCrossResponseNormalization(5,0.0001,0.75,1))

  local conv3 = backend.SpatialConvolution(256,384,3,3,1,1,1,1,1)
  conv3.name = 'conv3'
  shared:add(conv3)
  shared:add(backend.ReLU(true))

  local conv4 = backend.SpatialConvolution(384,384,3,3,1,1,1,1,2)
  conv4.name = 'conv4'
  shared:add(conv4)
  shared:add(backend.ReLU(true))

  local conv5 = backend.SpatialConvolution(384,256,3,3,1,1,1,1,2)
  conv5.name = 'conv5'
  shared:add(conv5)
  shared:add(backend.ReLU(true))

   -- Convolutions and roi info
  local shared_roi_info = nn.ParallelTable()
  shared_roi_info:add(shared)
  shared_roi_info:add(nn.Identity())

  -- Linear Part
  local linear = nn.Sequential()
  linear:add(nn.View(-1):setNumInputDims(3))
 
  local fc6 
  fc6 = nn.Linear(9216,4096) 
  fc6.name = 'fc6'
  linear:add(fc6)
  linear:add(backend.ReLU(true))
  linear:add(nn.Dropout(config.dropout)) --
 
  local fc7 = nn.Linear(4096,4096)
  fc7.name = 'fc7'
  linear:add(fc7)
  linear:add(backend.ReLU(true))
  linear:add(nn.Dropout(config.dropout)) --
 
  local ROIPoolingDA = detection.ROIPoolingDA(6,6):setSpatialScale(1/16)  
  local FeatFusion = detection.FeatureMapFusion(6,6,10) -- H,W,num_rois
 
  local model = nn.Sequential() 
  model:add(shared_roi_info) 
  model:add(ROIPoolingDA)
  model:add(FeatFusion)
  model:add(linear)
 
 local classifier = nn.Linear(4096+8,config.nclass+1) -- adding x_c,y_c,w,h for two boxes

  -- Whole Model
local splitter = nn.ParallelTable()
splitter:add(model) -- model returns  15 x 4096 fc7 features
splitter:add(nn.Identity()) -- it passes the 15x8 normalised coordinates

local model_main = nn.Sequential()
model_main:add(splitter)
model_main:add(nn.JoinTable(2))
model_main:add(classifier) 
 
  model_main:cuda()
  return model_main, fc6, fc7, classifier, name
   
end
return create_model
--------------------------------------------------------------------------------------------------------------
With this above network, I am able to run the forward pass successfully but getting the following error during the backward pass:

qlua: /home/suman/torch/install/share/lua/5.1/nn/Container.lua:67:
In 3 module of nn.Sequential:
/home/suman/torch/install/share/lua/5.1/nn/Linear.lua:86: invalid arguments: CudaTensor number number table CudaTensor
expected arguments: *CudaTensor~2D* [CudaTensor~2D] [float] CudaTensor~2D CudaTensor~2D | *CudaTensor~2D* float [CudaTensor~2D] float CudaTensor~2D CudaTensor~2D
stack traceback:
    [C]: in function 'addmm'
    /home/suman/torch/install/share/lua/5.1/nn/Linear.lua:86: in function 'updateGradInput'
    /home/suman/torch/install/share/lua/5.1/nn/Module.lua:31: in function </home/suman/torch/install/share/lua/5.1/nn/Module.lua:29>
    [C]: in function 'xpcall'
    /home/suman/torch/install/share/lua/5.1/nn/Container.lua:63: in function 'rethrowErrors'
    /home/suman/torch/install/share/lua/5.1/nn/Sequential.lua:84: in function 'backward'
    ./train/SequentialTrainerSS.lua:165: in function 'opfunc'
    /home/suman/torch/install/share/lua/5.1/optim/sgd.lua:44: in function 'sgd'
    ./train/SequentialTrainerSS.lua:175: in function '_trainBatch'
    ./train/SequentialTrainerSS.lua:117: in function 'train'
    ./network/NetworkWrapperSS.lua:47: in function 'trainNetwork'
    ...ode/ws-fast-rcnn-torch-mahyarnajibi/v2/main_train_da.lua:69: in main chunk

Please, someone provide some guidance.

**Comment by bamos** [comment]
Hi, I'm closing this issue because it is not an OpenFace issue or development discussion. Somebody on the OpenFace or Torch mailing lists might be able to help you.

-Brandon.
